google/cloud/bigtable/admin_v2/overlay/services/bigtable_table_admin/client.py

```diff
@@ -0,0 +1,199 @@
+
+                # Initialize request argument(s)
+                request = admin_v2.RestoreTableRequest(
+                    backup="backup_value",
+                    parent="parent_value",
+                    table_id="table_id_value",
+                )
+
+                # Make the request
+                operation = client.restore_table(request=request)
+
+                print("Waiting for operation to complete...")
+
+                response = operation.result()
+
+                # Handle the response
+                print(response)
+
+                # Handle LRO2
+                optimize_operation = operation.optimize_restore_table_operation()
+
+                if optimize_operation:
+                    print("Waiting for table optimization to complete...")
+
+                    response = optimize_operation.result()
+
+        Args:
+            request (Union[google.cloud.bigtable.admin_v2.types.RestoreTableRequest, dict]):
+                The request object. The request for
+                [RestoreTable][google.bigtable.admin.v2.BigtableTableAdmin.RestoreTable].
+            retry (google.api_core.retry.Retry): Designation of what errors, if any,
+                should be retried.
+            timeout (float): The timeout for this request.
+            metadata (Sequence[Tuple[str, Union[str, bytes]]]): Key/value pairs which should be
+                sent along with the request as metadata. Normally, each value must be of type `str`,
+                but for metadata keys ending with the suffix `-bin`, the corresponding values must
+                be of type `bytes`.
+
+        Returns:
+            google.cloud.bigtable.admin_v2.overlay.types.restore_table.RestoreTableOperation:
+                An object representing a long-running operation.
+
+                The result type for the operation will be :class:`google.cloud.bigtable.admin_v2.types.Table` A collection of user data indexed by row, column, and timestamp.
+                   Each table is served using the resources of its
+                   parent cluster.
+        """
+        operation = self._restore_table(
+            request=request,
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+        restore_table_operation = restore_table.RestoreTableOperation(
+            self._transport.operations_client, operation
+        )
+        return restore_table_operation
+
+    def await_consistency(
+        self,
+        request: Optional[
+            Union[bigtable_table_admin.CheckConsistencyRequest, dict]
+        ] = None,
+        *,
+        name: Optional[str] = None,
+        consistency_token: Optional[str] = None,
+        retry: OptionalRetry = gapic_v1.method.DEFAULT,
+        timeout: Union[float, object] = gapic_v1.method.DEFAULT,
+        metadata: Sequence[Tuple[str, Union[str, bytes]]] = (),
+    ) -> consistency.CheckConsistencyPollingFuture:
```

Comment:
docstring?

Can the word await here be avoided? That could be complicated for asyncio